### DISCUSSION TITLE
ch2/ch2-07

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#227] [General]
# ch2/ch2-07

https://golang-china.github.io/gopl-zh/ch2/ch2-07.html

**Comment by Apahno** [comment]
打卡

**Comment by ycs0220** [comment]
打卡

**Comment by AnhTom2000** [comment]
打卡

**Comment by HoronLee** [comment]
打卡

**Comment by wiwuwiwu** [comment]
打卡✅

**Comment by xqnjkycx** [comment]
var cwd string

func init() {
    cwd, err := os.Getwd() // NOTE: wrong!
    if err != nil {
        log.Fatalf("os.Getwd failed: %v", err)
    }
    log.Printf("Working directory = %s", cwd)
}

注意这里，短变量声明有赋值行为，为什么这里没有给包级变量cwd赋值，而是声明了一个局部变量？**这是因为短变量声明产生赋值行为的前提是，变量在同一作用域下**
